src/math/fibonacci.lua

```diff
@@ -0,0 +1,80 @@
+-- Fibonacci.lua
+
+-- Fibonacci Sequence:
+-- The Fibonacci sequence is a series of numbers where each number (after the first two) is the sum of the two preceding ones.
+-- More information: https://en.wikipedia.org/wiki/Fibonacci_number
+
+-- Author: Gyandeep
+
+--      Implemented nth Fibonacci number using different approaches in O(2^N) , O(N) , O(logN) , O(1) .
+
+
+-- Function to calculate Fibonacci numbers using recursion
+-- Time Complexity: O(2^n)
+function fibonacci_recursive(n)
+    if n <= 0 then
+        return 0
+    elseif n == 1 then
+        return 1
+    else
+        return fibonacci_recursive(n - 1) + fibonacci_recursive(n - 2)
+    end
+end
+
+-- Function to calculate Fibonacci numbers using dynamic programming (DP)
+-- Time Complexity: O(n)
+function fibonacci_dp(n)
+    local fib = {}
+    fib[0] = 0
+    fib[1] = 1
+    for i = 2, n do
+        fib[i] = fib[i - 1] + fib[i - 2]
+    end
+    return fib[n]
+end
+
+-- Function to calculate Fibonacci numbers using Binet's formula
+-- Time Complexity: O(1)
+-- Limitation: Accurate results up to n = 70 due to limitations of floating-point precision.
+function fibonacci_binet(n)
+    local phi = (1 + math.sqrt(5)) / 2
+    local psi = (1 - math.sqrt(5)) / 2
+    return math.floor((math.pow(phi, n) - math.pow(psi, n)) / math.sqrt(5))
+end
+
+-- Function to multiply two 2x2 matrices
+-- Time Complexity: O(1)
+function matrix_multiply(a, b)
```

Comment:
Ideally this should not need to be here if I had finished and pushed my matrix "class" yet :P

Reply:
https://github.com/TheAlgorithms/Lua/commit/3ca92fb7781334a020fb86a56b4acaa12b27c064 finished and pushed it